### PR TITLE
Raster layer file

### PR DESCRIPTION
**Pull request by saratchandrakarumuri**

### CODE REVIEW

File: arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        self.rasterLayer = AGSRasterLayer(raster: raster)
+        
+        //initialize map with raster layer as the basemap
+        self.map = AGSMap(basemap: AGSBasemap.imagery())
+        
+        self.mapView.map = map
+        
+        self.mapView.map?.operationalLayers.add(rasterLayer!)
+        
+        self.rasterLayer.addObserver(self, forKeyPath: "loadStatus", options: .new, context: nil)
```
Comment (dg0yal) [inline arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift]: Probably a better way to do this is to call load() on the raster layer, and in the completion block set the viewpoint. KVO feels a little more verbose for this which can be easily accomplished inline using  a block

**Comment by saratchandrakarumuri** [comment]
@dg0yal Updated the sample to use completion block instead of KVO

**Comment by dg0yal** [comment]
Looks good, But looks like your merge is deleting a lot of new samples/code that was recently added. You need to update your master branch, and then update your RasterLayerFile with those changes so they don't wipe anything out

**Comment by saratchandrakarumuri** [comment]
@dg0yal Updated to match changes in v.next branch

File: arcgis-ios-sdk-samples.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1695,7 +1710,9 @@
 				3ED028A71B8E3A9800ACA70D /* Change sublayer visibility */,
 				3EE8B9511CA08E16003AA705 /* ArcGIS vector tile layer (URL) */,
 				3E81C1B71D8A0C7400470AA4 /* ArcGIS vector tiled layer (custom style) */,
+				166E8F951EE4DF6C00C57C46 /* Raster layer (file) */,
 				3E6294081EAFAF4100084105 /* Web tiled layer */,
+				166E8F951EE4DF6C00C57C46 /* Raster layer (file) */,
```
Comment (dg0yal) [inline arcgis-ios-sdk-samples.xcodeproj/project.pbxproj]: why are there two of these?

File: arcgis-ios-sdk-samples/Layers/Raster layer (file)/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Raster layer (file)
+
+This sample demonstrates how to create and use a raster layer made from a local raster file.
+
+![](image1.png)
+
+## How it works
+
+`AGSRasterLayer` provides an initializer called `init(raster:)` that takes a `AGSRaster`.
```
Comment (dg0yal) [inline arcgis-ios-sdk-samples/Layers/Raster layer (file)/README.md]: Describe how you create AGSRaster to use a file on disk

File: arcgis-ios-sdk-samples.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -134,6 +137,9 @@
 		3E85263D1B90DE2C00690EBD /* DisplayMapViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3E85263C1B90DE2C00690EBD /* DisplayMapViewController.swift */; };
 		3E85263F1B90E06F00690EBD /* DisplayMap.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 3E85263E1B90E06F00690EBD /* DisplayMap.storyboard */; };
 		3E8526401B90E19C00690EBD /* DisplayMapViewController.swift in CopyFiles */ = {isa = PBXBuildFile; fileRef = 3E85263C1B90DE2C00690EBD /* DisplayMapViewController.swift */; };
+		3E8A05BA1ECD07DD0032231C /* SceneLayerURL.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 3E8A05B91ECD07DD0032231C /* SceneLayerURL.storyboard */; };
```
Comment (dg0yal) [inline arcgis-ios-sdk-samples.xcodeproj/project.pbxproj]: The order of some scene layer sample files have been changed in the project file. Don't know if this has an impact on the order in which that sample is displayed in the app. Can you check that out and see if this change needs to be reverted, or if it is harmless to the app.

File: arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        
+        //create raster layer using raster
+        self.rasterLayer = AGSRasterLayer(raster: raster)
+        
+        //initialize map with raster layer as the basemap
+        self.map = AGSMap(basemap: AGSBasemap.imagery())
+        
+        self.mapView.map = map
+        
+        self.mapView.map?.operationalLayers.add(rasterLayer!)
```
Comment (dg0yal) [inline arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift]: add comment

File: arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        
+        self.rasterLayer.load { (error) in
```
Comment (dg0yal) [inline arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift]: add comment on what this is doing

**Comment by saratchandrakarumuri** [comment]
@dg0yal Made changes as per your feedback

File: arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift (hunk 6 above)
Comment (singhgag) [inline arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift]: @saratchandrakarumuri You are strongly capturing self here. Could result in a reference cycle.

File: arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift [hunk 7]
```diff
@@ -42,15 +42,11 @@ class RasterLayerVC: UIViewController {
...
-    }
-
-    
-    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
-        if self.rasterLayer.loadStatus == AGSLoadStatus.loaded {
-            self.mapView.setViewpoint(AGSViewpoint(center: (self.rasterLayer.fullExtent?.center)!, scale: 80000))
-        }
+        self.rasterLayer.load { (error) in
+            if error == nil {
+                self.mapView.setViewpoint(AGSViewpoint(center: (self.rasterLayer.fullExtent?.center)!, scale: 80000))
```
Comment (singhgag) [inline arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift]: Add weak reference to self and update this line of code to remove implicit unwrap. Use optional binding instead.

File: arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift [hunk 8]
```diff
@@ -42,15 +42,11 @@ class RasterLayerVC: UIViewController {
...
-        
-    }
-
-    
-    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
-        if self.rasterLayer.loadStatus == AGSLoadStatus.loaded {
-            self.mapView.setViewpoint(AGSViewpoint(center: (self.rasterLayer.fullExtent?.center)!, scale: 80000))
-        }
+        self.rasterLayer.load { (error) in
+            if error == nil {
```
Comment (singhgag) [inline arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift]: For error, use SVProgressHUD to show error to the user.

**Comment by singhgag** [comment]
@saratchandrakarumuri Merged. Good work!

**Comment by dg0yal** [comment]
Congrats @saratchandrakarumuri  🎉 🎈 🎉

**Pull request merged**